### PR TITLE
PR #1 : Github Actions, Docker, Environmental Variables

### PR DESCRIPTION
**Pull request by yashcode00**
🚀 **PR 1: Summary:**

The following are the things done in this PR:

1. Added updated and working Docker files for both server and indexer.
2. Implemented loading code for confidential and sensitive information using environment variables instead of the `conf.hcl` file.
3. Added some more environment variables and removed them from constants.
4. Enabled service account usage for the server.
5. Completed and implemented a GitHub Actions pipeline (CI/CD) that currently uses personal DockerHub details to push the Docker image.
6. Made `conf.hcl` public as it no longer contains any confidential information and made minor staging process changes.
7. Created a functional Docker Compose file with all necessary containers, including the server, indexer, and Postgres database.

### CODE REVIEW

File: internal/indexer/indexer.go [hunk 1]
```diff
@@ -429,7 +429,11 @@ func (idx *Indexer) Run() error {
 				logError("error exporting document", err)
 				os.Exit(1)
 			}
+
 			content, err := io.ReadAll(exp.Body)
+
+			fmt.Println("Before trimming: ", content, " ---->> lengthh: ", len(content))
```
Comment (ashisht99x) [inline internal/indexer/indexer.go:435]: not needed

File: internal/indexer/indexer.go [hunk 2]
```diff
@@ -439,6 +443,8 @@ func (idx *Indexer) Run() error {
 				content = content[:maxContentSize]
 			}
 
+			fmt.Println("After trimming: ", content, " ---->> lengthh: ", len(content))
```
Comment (ashisht99x) [inline internal/indexer/indexer.go:446]: not needed

File: pkg/googleworkspace/service.go [hunk 3]
```diff
@@ -176,8 +179,10 @@ func getTokenFromWeb(config *oauth2.Config) *oauth2.Token {
 
 	m := http.NewServeMux()
 	// TODO: remove hardcoded port.
-	s := http.Server{Addr: ":9999", Handler: m}
-	config.RedirectURL = "http://localhost:9999/callback"
+	// s := http.Server{Addr: ":9999", Handler: m}
+	// config.RedirectURL = "http://localhost:9999/callback"
+	s := http.Server{Addr: ":9000", Handler: m}
+	config.RedirectURL = "https://doc-vault.dev.razorpay.in/callback"
```
Comment (ashisht99x) [inline pkg/googleworkspace/service.go:185]: should be passed as environment variable ?

Comment (yashcode00) [reply pkg/googleworkspace/service.go:185]: they have not provided this facility, we might add it, but actually, it is not required as we have now shifted to **service account**, this part of the `code `was its alternative.

File: internal/indexer/indexer.go (hunk 2 above)
Comment (yashcode00) [reply internal/indexer/indexer.go]: done

File: internal/indexer/indexer.go (hunk 1 above)
Comment (yashcode00) [reply internal/indexer/indexer.go]: done

**Review by ashisht99x — APPROVED** [review]

**Pull request merged**